src/overlay/Peer.cpp

```diff
@@ -1078,9 +1080,81 @@ Peer::recvDontHave(StellarMessage const& msg)
 }
 
 void
-Peer::recvGetTxSet(StellarMessage const& msg)
+Peer::sendTxSet(TxSetFrameConstPtr txSet)
+{
+    StellarMessage newMsg;
+    if (txSet->isGeneralizedTxSet())
+    {
+        if (mRemoteOverlayVersion <
+            Peer::FIRST_VERSION_SUPPORTING_GENERALIZED_TX_SET)
+        {
+            // The peer wouldn't be able to accept the generalized tx set,
+            // but it wouldn't be correct to say we don't have it. So we
+            // just let the request to timeout.
+            return;
+        }
+        newMsg.type(GENERALIZED_TX_SET);
+        txSet->toXDR(newMsg.generalizedTxSet());
+    }
+    else
+    {
+        newMsg.type(TX_SET);
+        txSet->toXDR(newMsg.txSet());
+    }
+
+    auto newMsgPtr = std::make_shared<StellarMessage const>(newMsg);
+    sendMessage(newMsgPtr);
+}
+
+//                 ┏━━━━━━━━━━━━━━━━━┓
+//                 ┃                 ┃     Retry
+//                 ┃                 ┃    ┌after ────────┐
+//      ┌──────────┃Txn set fetching ┃    timeout.       │
+//      │          ┃                 ┃    │              │
+//      │          ┃                 ┃    │     ┌────────────────┐
+//      │          ┗━━━━━━━━━━━━━━━━━┛    │     │Add remote peer │
+//      │                   │             │     │   to pending   │
+//      │                   │             │     │    getTxSet    │
+//      │                   │             │     │requests for tx │
+//      │                   │             │     └────────────────┘
+//      │          ┌────────┴───────┐     │              ▲
+// Local node      │Local node recvs│     │              │
+//  receives       │   GetTxnSet    │◀────┘              │
+//  new txn        │    request.    │                   Y│
```

Comment:
What does "Retry after timeout" mean here? It would be the remote node retrying, right?